rpc/src/services/base_services.rs

```diff
@@ -23,9 +23,15 @@ use tezos_messages::ts_to_rfc3339;
 
 pub type BlockOperationsHashes = Vec<String>;
 
+use cached::proc_macro::cached;
+use cached::TimedCache;
+use cached::TimedSizedCache;
+use cached::SizedCache;
+
 /// Retrieve blocks from database.
+#[cached( name="BLOCK_HASH_CACHE",type = "TimedCache<(ChainId,BlockHash), Vec<BlockHash>>", create = "{TimedCache::with_lifespan(20)}", convert = "{(chain_id.clone(),block_hash.clone())}", result = true)]
 pub(crate) fn get_block_hashes(
-    _chain_id: ChainId,
+    chain_id: ChainId,
```

Comment:
@mambisi 
to aviod this unused warning,
just leave it like before:
```_chain_id: ChainId```

and just in that annotation use:
```{(_chain_id.clone()```